### PR TITLE
fix(l10n): directly import translate functions to the components

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

Scripted and added manually

Check with:
```php
<?php
function injectStrings($directory) {
	$iterator = new RecursiveIteratorIterator(
		new RecursiveDirectoryIterator($directory),
		RecursiveIteratorIterator::SELF_FIRST
	);
	$countT = 0;
	$countN = 0;

	foreach ($iterator as $file) {
		if ($file->isFile()) {
			$filePath = $file->getPathname();
			$contents = file_get_contents($filePath);

			if (str_contains($filePath, ".php")) {
				// not needed
				continue;
			}

			if (str_contains($contents, "t('spreed'")) {
				$countT += 1;

				if (str_contains($filePath, "mainFilesSidebarLoader.js")) {
					// it has init.js file with import
					continue;
				}

				if (!str_contains($contents, "import { t, n } from '@nextcloud/l10n'\n")
				&& !str_contains($contents, "import { t } from '@nextcloud/l10n'\n")
				&& !str_contains($contents, "import { t, getLocale } from '@nextcloud/l10n'\n")
				&& !str_contains($contents, "import { n } from '@nextcloud/l10n'\n")) {
					echo "Missing pattern in $filePath\n";
				}

				if (str_contains($filePath, ".vue")
					&& !str_contains($contents, "methods: {\n")
					&& !str_contains($contents, "\n		t,")) {
					echo "Vue file missing method $filePath\n";
				}
			}
			if (str_contains($contents, "n('spreed'")) {
				$countN += 1;

				if (!str_contains($contents, "import { t, n } from '@nextcloud/l10n'\n")
					&& !str_contains($contents, "import { t } from '@nextcloud/l10n'\n")
					&& !str_contains($contents, "import { t, getLocale } from '@nextcloud/l10n'\n")
					&& !str_contains($contents, "import { n } from '@nextcloud/l10n'\n")) {
					echo "Missing pattern in $filePath\n";
				}

				if (str_contains($filePath, ".vue")
					&& !str_contains($contents, "methods: {\n")
					&& !str_contains($contents, "\n		n,")) {
					echo "Vue file missing method $filePath\n";
				}
			}
		}
	}

	echo "$countT files in total\n";
	echo "$countN plural files in total\n\n";
}

injectStrings('.');
?>
```

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [x] 🖥️ Tested with Desktop client or should not be risky for it 
- [x] ⛑️ Tests are included or not possible

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -25,6 +25,7 @@ import { provide } from 'vue'
 
 import { getCurrentUser } from '@nextcloud/auth'
 import { emit, subscribe, unsubscribe } from '@nextcloud/event-bus'
+import { translate as t } from '@nextcloud/l10n'
```
Comment (ShGKme) [inline src/App.vue:28]:
```suggestion
import { t } from '@nextcloud/l10n'
```

**Comment by ShGKme** [comment]
Shall we do it in `main`?

**Comment by Antreesy** [comment]
> Shall we do it in `main`?

There's a lot of files conflicts, so I'd keep it as a `next` version, what do you think?

**Review by ShGKme — APPROVED** [review]

**Comment by ShGKme** [comment]
For history, using global `t` wasn't good because:
- It's indeterminate. There is no guarantee, `window.t` is the same version as installed `@nextcloud/l10n`. Different modules may use different versions. Same with `Vue.prototype.t` in Vue 2.
- No way to check if in this context `t` is defined. Either error from ESLint/TS everywhere, or missing actual errors. We have problems from that in `@nextcloud/vue` time to time

**Review by DorraJaouad — APPROVED** [review]
🦅

**Pull request merged**